samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -405,78 +402,131 @@ BT_CONN_CB_DEFINE(conn_callbacks) = {
 
 static void fp_adv_provisioning_state_changed(bool provisioned)
 {
+	struct bt_le_oob oob;
+	int err;
+
 	fmdn_provisioned = provisioned;
+
+	if (!provisioned) {
+		/* Force the RPA rotation to synchronize the Fast Pair advertising
+		 * payload with its RPA address using rpa_expired callback.
+		 */
+		err = bt_le_oob_get_local(fp_adv_param.id, &oob);
+		if (err) {
+			LOG_ERR("Fast Pair: bt_le_oob_get_local failed: %d", err);
+		}
+	}
 }
 
 static struct bt_fast_pair_fmdn_info_cb fmdn_info_cb = {
 	.provisioning_state_changed = fp_adv_provisioning_state_changed,
 };
 
-void app_fp_adv_mode_set(enum app_fp_adv_mode adv_mode)
+enum app_fp_adv_mode app_fp_adv_mode_get(void)
+{
+	return fp_adv_mode;
+}
+
+static int fp_adv_trigger_idx_get(struct app_fp_adv_trigger *trigger)
```

Comment:
let's use the assert version from before. After all, you simply assert the error code in the app_fp_adv_request